src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/DesiredOutcomeRepository.kt

```diff
@@ -4,4 +4,6 @@ import org.springframework.data.jpa.repository.JpaRepository
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.DesiredOutcome
 import java.util.UUID
 
-interface DesiredOutcomeRepository : JpaRepository<DesiredOutcome, UUID>
+interface DesiredOutcomeRepository : JpaRepository<DesiredOutcome, UUID> {
+  fun findByServiceCategoryId(serviceCategoryId: UUID): List<DesiredOutcome>
```

Comment:
Would we also use this in a production-path scenario? Just thinking if this is a hint towards something deeper if we find it useful for contract testing

Reply:
i wrote this with somewhat bleary eyes - i think the correct way to get the desired outcomes is just `serviceCategoryRepository.findById({id}).desiredOutcomes`